lib/davinci_pdex_test_kit/pdex_payer_client/client_registration_group.rb

```diff
@@ -0,0 +1,67 @@
+require 'udap_security_test_kit'
+require 'smart_app_launch_test_kit'
+require_relative 'pdex_client_options'
+require_relative 'client_registration/configuration_display_smart_test'
+require_relative 'client_registration/configuration_display_udap_test'
+
+module DaVinciPDexTestKit
+  module PDexPayerClient
+    class PDexClientRegistrationGroup < Inferno::TestGroup
+      id :pdex_client_registration
+      title 'Client Registration'
+      description %(
+        Register the client under test with Inferno's simulated PDex Server,
+        including configuration of the system under test to hit the correct endpoints and
+        enable authentication and authorization of PDex requests.
+      )
+      run_as_group
+
+      # smart registration tests
+      test from: :smart_client_registration_alca_verification,
+            required_suite_options: {
+              client_type: PDexClientOptions::SMART_APP_LAUNCH_CONFIDENTIAL_ASYMMETRIC
+            }
+      test from: :smart_client_registration_alcs_verification,
+            required_suite_options: {
+              client_type: PDexClientOptions::SMART_APP_LAUNCH_CONFIDENTIAL_SYMMETRIC
+            }
+      test from: :smart_client_registration_alp_verification,
+            required_suite_options: {
+              client_type: PDexClientOptions::SMART_APP_LAUNCH_PUBLIC
+            }
+      test from: :pdex_client_reg_config_smart_display,
+            id: :pdex_client_reg_config_smart_alca_display,
+            required_suite_options: {
+              client_type: PDexClientOptions::SMART_APP_LAUNCH_CONFIDENTIAL_ASYMMETRIC
+            }
+      test from: :pdex_client_reg_config_smart_display,
+            id: :pdex_client_reg_config_smart_alcs_display,
+            required_suite_options: {
+              client_type: PDexClientOptions::SMART_APP_LAUNCH_CONFIDENTIAL_SYMMETRIC
+            }
+      test from: :pdex_client_reg_config_smart_display,
+            id: :pdex_client_reg_config_smart_alp_display,
+            required_suite_options: {
+              client_type: PDexClientOptions::SMART_APP_LAUNCH_PUBLIC
+            }
```

Comment:
Is there a reason these need to be separated by client type? I don't see any configurations being changed in each of these so these may not need to be separated

Reply:
Yes, because they can't be present when the UDAP option is chosen, but there's no way to say "include if one of these three suite options is selected". So in this case, you have to have it 3 times, once for each suite option that does need it.

Reply:
Ah right I completely forgot about that, okay never mind I will resolve this issue!